File: runtime/flutter/foundation/flutterError.ts
import { IAssertionError } from "../../dart/core/assertionError";
import { IError } from "../../dart/core/error";
import { IIterable } from "../../dart/core/iterable";
import { IList } from "../../dart/core/list";
import { IStackTrace } from "../../dart/core/stackTrace";
import { IDiagnosticable } from "./diagnosticable";
import { IDiagnosticableTree } from "./diagnosticableTree";
import { IDiagnosticableTreeMixin } from "./diagnosticableTreeMixin";
import { DiagnosticLevel } from "./diagnosticLevel";
import { IDiagnosticPropertiesBuilder } from "./diagnosticPropertiesBuilder";
import { IDiagnosticsNode } from "./diagnosticsNode";
import { DiagnosticsTreeStyle } from "./diagnosticsTreeStyle";
import { IFlutterErrorDetails } from "./flutterErrorDetails";
import { IStackFilter } from "./stackFilter";
declare const flutter: {
    foundation: {
        flutterError: (
            this: void,
            flutterError: IFlutterError,
            message: string
        ) => IFlutterError;
        flutterErrorResetErrorCount: () => void;
        flutterErrorDumpErrorToConsole: (
            details: IFlutterErrorDetails,
            props: { forceReport: boolean }
        ) => void;
        flutterErrorAddDefaultStackFilter: (filter: IStackFilter) => void;
        flutterErrorDefaultStackFilter: (
            frames: IIterable<string>
        ) => IIterable<string>;
        flutterErrorReportError: (details: IFlutterErrorDetails) => void;
    };
};
export interface IFlutterError {
    diagnostics: IList<IDiagnosticsNode>;
    message: Object | undefined;
    getMessage: () => string;
    debugFillProperties: (properties: IDiagnosticPropertiesBuilder) => void;
    toStringShort: () => string;
    toString: (props: { minLevel: DiagnosticLevel }) => string;
    toStringShallow: (props: {
        joiner: string;
        minLevel: DiagnosticLevel;
    }) => string;
    toStringDeep: (props: {
        minLevel: DiagnosticLevel;
        prefixLineOne: string;
        prefixOtherLines?: string | undefined;
    }) => string;
    toDiagnosticsNode: (props?: {
        name?: string | undefined;
        style?: DiagnosticsTreeStyle | undefined;
    }) => IDiagnosticsNode;
    debugDescribeChildren: () => IList<IDiagnosticsNode>;
    getStackTrace: () => IStackTrace | undefined;
    getHashCode: () => number;
}
export class FlutterError
    implements
        IError,
        IDiagnosticableTreeMixin,
        IDiagnosticable,
        IAssertionError,
        Omit<
            IDiagnosticableTree,
            | "toStringShallow"
            | "toStringDeep"
            | "debugDescribeChildren"
            | "debugFillProperties"
            | "toStringShort"
            | "toDiagnosticsNode"
        >
{
    public static wrapWidth = 100;
    public readonly diagnostics: IList<IDiagnosticsNode> = undefined as any;
    public readonly message: Object | undefined = undefined as any;
    public constructor(message: string) {
        flutter.foundation.flutterError(this, message);
    }
    public static resetErrorCount(): void {
        return flutter.foundation.flutterErrorResetErrorCount();
    }
    public static dumpErrorToConsole(
        details: IFlutterErrorDetails,
        props: { forceReport?: boolean }
    ): void {
        return flutter.foundation.flutterErrorDumpErrorToConsole(details, {
            ...dumpErrorToConsoleDefaultProps,
            ...props,
        });
    }
    public static addDefaultStackFilter(filter: IStackFilter): void {
        return flutter.foundation.flutterErrorAddDefaultStackFilter(filter);
    }
    public static defaultStackFilter(
        frames: IIterable<string>
    ): IIterable<string> {
        return flutter.foundation.flutterErrorDefaultStackFilter(frames);
    }
    public static reportError(details: IFlutterErrorDetails): void {
        return flutter.foundation.flutterErrorReportError(details);
    }
    private readonly _dart_getMessage: () => string = undefined as any;
    private readonly _dart_debugFillProperties: (
        properties: IDiagnosticPropertiesBuilder
    ) => void = undefined as any;
    private readonly _dart_toStringShort: () => string = undefined as any;
    private readonly _dart_toString: (props: {
        minLevel: DiagnosticLevel;
    }) => string = undefined as any;
    private readonly _dart_toStringShallow: (props: {
        joiner: string;
        minLevel: DiagnosticLevel;
    }) => string = undefined as any;
    private readonly _dart_toStringDeep: (props: {
        minLevel: DiagnosticLevel;
        prefixLineOne: string;
        prefixOtherLines?: string | undefined;
    }) => string = undefined as any;
    private readonly _dart_toDiagnosticsNode: (props?: {
        name?: string | undefined;
        style?: DiagnosticsTreeStyle | undefined;
    }) => IDiagnosticsNode = undefined as any;
    private readonly _dart_debugDescribeChildren: () => IList<IDiagnosticsNode> =
        undefined as any;
    private readonly _dart_getStackTrace: () => IStackTrace | undefined =
        undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    public getMessage(): string {
        return this._dart_getMessage();
    }
    public debugFillProperties(properties: IDiagnosticPropertiesBuilder): void {
        return this._dart_debugFillProperties(properties);
    }
    public toStringShort(): string {
        return this._dart_toStringShort();
    }
    public toString(props: { minLevel?: DiagnosticLevel }): string {
        return this._dart_toString({
            ...toStringDefaultProps,
            ...props,
        });
    }
    public toStringShallow(props: {
        joiner?: string;
        minLevel?: DiagnosticLevel;
    }): string {
        return this._dart_toStringShallow({
            ...toStringShallowDefaultProps,
            ...props,
        });
    }
    public toStringDeep(props: {
        minLevel?: DiagnosticLevel;
        prefixLineOne?: string;
        prefixOtherLines?: string | undefined;
    }): string {
        return this._dart_toStringDeep({
            ...toStringDeepDefaultProps,
            ...props,
        });
    }
    public toDiagnosticsNode(props?: {
        name?: string | undefined;
        style?: DiagnosticsTreeStyle | undefined;
    }): IDiagnosticsNode {
        return this._dart_toDiagnosticsNode(props);
    }
    public debugDescribeChildren(): IList<IDiagnosticsNode> {
        return this._dart_debugDescribeChildren();
    }
    public getStackTrace(): IStackTrace | undefined {
        return this._dart_getStackTrace();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
}
const toStringDefaultProps = {
    minLevel: DiagnosticLevel.info,
};
const toStringShallowDefaultProps = {
    joiner: ", ",
    minLevel: DiagnosticLevel.debug,
};
const toStringDeepDefaultProps = {
    minLevel: DiagnosticLevel.debug,
    prefixLineOne: "",
};
const dumpErrorToConsoleDefaultProps = {
    forceReport: false,
};

File: lib/cfr/builtins/libs/flutter/foundation/flutterError.dart
import 'dart:core' as _fac9;

import 'package:flutter/src/foundation/assertions.dart' as _5dcc;
import 'package:flutter/src/foundation/diagnostics.dart' as _eaf3;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedFlutterError extends _36c2.VMManagedBox<_5dcc.FlutterError> {VMManagedFlutterError({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'diagnostics'
] = _36c2.maybeBoxObject<_fac9.List<_eaf3.DiagnosticsNode>>(object: vmObject.diagnostics, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'message'
] = _36c2.maybeBoxObject<_fac9.Object?>(object: vmObject.message, hydroState: hydroState, table: _36c2.HydroTable());
table  [
'getMessage'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.message,]; } );
table  [
'debugFillProperties'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.debugFillProperties(_36c2.maybeUnBoxAndBuildArgument<_eaf3.DiagnosticPropertiesBuilder, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'toStringShort'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toStringShort(),]; } );
table  [
'toString'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.toString(minLevel: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticLevel.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'minLevel'
] : null)),]; } );
table  [
'toStringShallow'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [vmObject.toStringShallow(joiner: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'joiner'
] : null, minLevel: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticLevel.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'minLevel'
] : null)),]; } );
table  [
'toStringDeep'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 

return [vmObject.toStringDeep(minLevel: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticLevel.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'minLevel'
] : null), prefixLineOne: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'prefixLineOne'
] : null, prefixOtherLines: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'prefixOtherLines'
] : null),]; } );
table  [
'toDiagnosticsNode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return [_36c2.maybeBoxObject<_eaf3.DiagnosticsNode>(object: vmObject.toDiagnosticsNode(name: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'name'
] : null, style: _36c2.maybeUnBoxEnum(values: _eaf3.DiagnosticsTreeStyle.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'style'
] : null)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'debugDescribeChildren'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_fac9.List<_fac9.dynamic>>(object: vmObject.debugDescribeChildren().map((x) => _36c2.maybeBoxObject<_eaf3.DiagnosticsNode>(object: x, hydroState: hydroState, table: _36c2.HydroTable())).toList(), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'getStackTrace'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { final returnValue = vmObject.stackTrace;if(returnValue!= null){return [_36c2.maybeBoxObject<_fac9.StackTrace?>(object: returnValue, hydroState: hydroState, table: _36c2.HydroTable()),];}return []; } );
table  [
'getHashCode'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.hashCode,]; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _5dcc.FlutterError vmObject;

 }

void loadFlutterError({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { table  [
'flutterError'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
return  [_36c2.maybeBoxObject<_5dcc.FlutterError>(object: _5dcc.FlutterError(luaCallerArguments  [
1
]), hydroState: hydroState, table: luaCallerArguments  [
0
])]; } );
table  [
'flutterErrorResetErrorCount'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { _5dcc.FlutterError.resetErrorCount();
return []; } );
table  [
'flutterErrorDumpErrorToConsole'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
_5dcc.FlutterError.dumpErrorToConsole(_36c2.maybeUnBoxAndBuildArgument<_5dcc.FlutterErrorDetails, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), forceReport: luaCallerArguments.length >= 3 ? luaCallerArguments  [
2
]  [
'forceReport'
] : null);
return []; } );
table  [
'flutterErrorAddDefaultStackFilter'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { _5dcc.FlutterError.addDefaultStackFilter(_36c2.maybeUnBoxAndBuildArgument<_5dcc.StackFilter, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'flutterErrorDefaultStackFilter'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [_36c2.maybeBoxObject<_fac9.Iterable>(object: _5dcc.FlutterError.defaultStackFilter(_36c2.maybeUnBoxAndBuildArgument<_fac9.Iterable<_fac9.String>, _fac9.String>(luaCallerArguments  [
1
], parentState: hydroState)), hydroState: hydroState, table: _36c2.HydroTable()),]; } );
table  [
'flutterErrorReportError'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { _5dcc.FlutterError.reportError(_36c2.maybeUnBoxAndBuildArgument<_5dcc.FlutterErrorDetails, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
_36c2.registerBoxer<_5dcc.FlutterError>(boxer: ({required _5dcc.FlutterError vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedFlutterError(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
